lib/good_job/notifier.rb

```diff
@@ -165,6 +166,8 @@ def create_executor
     end
 
     def listen(delay: 0)
+      return unless @enable_listening
```

Comment:
This can't be an early return. Creating the future is necessary because the `Notifier` is a bit more of a general purpose Reactor now with more responsibilities than _just_ LISTENing e.g. this background thread creates/manages the `Process` record.

It will be messier but I think the `unless @enable_listening` needs to go within the Future/thread:

1. Calling `connection.execute("LISTEN #{CHANNEL}")` (and the ActiveSupport Notification wrapping it (though we still need the callback trigger
2. Within `#wait_for_notify` having `!@enable_listening` that cause the logic to fall down to the `sleep WAIT_INTERVAL` path
3. Calling `connection.execute("UNLISTEN *")`

Reply:
Thanks for explaining, I've added a couple commits to do the above and also tried avoiding checking out the connection. I'm not sure if it's right, and I couldn't figure out how to test it. `notifier.connection` was always `nil` whether or not `enable_listening` was enabled.